File: stack/_old/stack.js
class Stack {
  constructor() {
    this.data = [];
  }

  push(val) {
    this.data.push(val);
  }

  pop() {
    return this.data.pop();
  }

  peek() {
    return this.data[this.data.length - 1];
  }
}

const s = new Stack();
s.push(1); // [1]
s.push(2); // [1, 2]
s.peek(); // 2
s.pop(); // 2

File: stack/stack.py
class Stack:
    def __init__(self):
        self.data = []

    def push(self, val):
        self.data.append(val)
        return self.data

    def pop(self):
        self.data.pop()
        return self.data

    def peek(self):
        return self.data[-1]


stack = Stack()
print(stack.push(1))  # [1]
print(stack.push(2))  # [1, 2]
print(stack.peek())  # 2
print(stack.pop())  # [1]
